src/framework/components/anim/component-layer.js

```diff
@@ -233,9 +255,35 @@ class AnimComponentLayer {
     }
 
     update(dt) {
+        if (this._blendTime) {
+            if (this._blendTimeElapsed < this._blendTime) {
+                this.weight = math.lerp(this._startingWeight, this._targetWeight, this._blendTimeElapsed / this._blendTime);
+                this._blendTimeElapsed += dt;
+            } else {
+                this.weight = this._targetWeight;
+                this._blendTime = null;
+                this._blendTimeElapsed = 0;
+                this._startingWeight = null;
+                this._targetWeight = null;
```

Comment:
personally, if something is a number, I only store numbers to it and not null.
I suspect when type changes, the function needs recompile under the hood to run optimized.
I'd suggest to use numbers .. 0 would work just fine.

Reply:
Done 👍